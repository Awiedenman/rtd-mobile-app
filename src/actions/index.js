import { userInfo } from "os";

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
});
   
export const hasErrored = (bool) => ({
  type: 'HAS_ERRORED',
  hasErrored: bool
});

export const storeScheduleData = (schedule) => ({
  type: 'STORE_SCHEDULE_DATA',
  schedule
});

export const storeStartSearch = (startAddress) => ({
  type: 'STORE_START_SEARCH',
  startAddress
});

export const storeEndSearch = (endAddress) => ({
  type: 'STORE_END_SEARCH',
  endAddress
});

export const storeMockFavRoutes = (mockFavRoutes) => ({
  type: 'STORE_MOCK_FAV_ROUTES',
  mockFavRoutes
});

export const signUpUser = (userInfo) => ({
  type: 'SIGN_UP_USER',
  userInfo
});