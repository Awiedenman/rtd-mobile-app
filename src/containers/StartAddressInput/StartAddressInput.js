import React, { Component } from 'react';
import PlacesAutocomplete from 'react-places-autocomplete';
import { storeStartAddress } from '../../actions';
import { connect } from "react-redux";
import PropTypes from 'prop-types';

import '../Search/Search.css';
 
export class StartAddressInput extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      startAddress: ''
    };
  }
 
  handleChange = startAddress => {
    this.setState({ startAddress });
  };
 
  handleSelect = address => {
    this.props.storeStartAddress(address);
    this.setState({startAddress: address});
  };
 
  render() {
    return (
      <PlacesAutocomplete
        className='start-address-input' 
        name='startAddress'
        value={this.state.startAddress}
        onChange={this.handleChange}
        onSelect={this.handleSelect}
      >
        {({ getInputProps, suggestions, getSuggestionItemProps, loading }) => {
  
          return (
            <div>
              <input
                value={this.state.startAddress}
                {...getInputProps({
                  placeholder: 'Search Starting Point ...',
                  className: 'location-search-input'
                })}
              />
              <div className="autocomplete-dropdown-container">
                {loading && <div>Loading...</div>}
                {suggestions.map((suggestion, index )=> {
                  const className = suggestion.active
                    ? 'suggestion-item--active'
                    : 'suggestion-item';
                  const style = suggestion.active
                    ? { backgroundColor: '#fafafa', cursor: 'pointer' }
                    : { backgroundColor: '#ffffff', cursor: 'pointer' };
                  return (
                    <div
                      key={index}
                      {...getSuggestionItemProps(suggestion, {
                        className,
                        style
                      })}
                    >
                      <span>{suggestion.description}</span>
                    </div>
                  );
                })}
              </div>
            </div>
          );
        }}
      </PlacesAutocomplete>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  storeStartAddress: (startAddress) => dispatch(storeStartAddress(startAddress))
});

export default connect(null, mapDispatchToProps)(StartAddressInput);

StartAddressInput.propTypes = {
  storeStartAddress: PropTypes.func
};