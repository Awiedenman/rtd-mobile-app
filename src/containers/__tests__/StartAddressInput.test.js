import React from 'react';
import { StartAddressInput } from '../StartAddressInput/StartAddressInput';
import { shallow } from 'enzyme'; 
import { storeEndAddress } from '../../actions';

describe('StartAddressInput', () => {
  it('should match snapshot', () => {
    const wrapper =  shallow(
      <StartAddressInput/>);

    expect(wrapper).toMatchSnapshot();
  });
  
  describe('handleChange', () => {
    it('should update the state of "startAddress" on change of the input field', () => {
      const wrapper = shallow(<StartAddressInput />);

      wrapper.instance().handleChange('');

      expect(wrapper.state('startAddress')).toEqual('unions station');
    });
  });

  describe('handleSelect', () => {
    it('should call storeAddress on the select of the input form ', () => {
      const wrapper = shallow(<StartAddressInput />);
      const handleSelect = jest.fn();
      const { storeStartAddress } = jest.fn();
      const event = {
        target : {
          name: 'startAdress',
          value: 'union station'
        }
      };

      handleSelect();

      wrapper.find('.start-address-input').simulate('select', event);

      expect(storeStartAddress).toHaveBeenCalled();
    });
  });
});
