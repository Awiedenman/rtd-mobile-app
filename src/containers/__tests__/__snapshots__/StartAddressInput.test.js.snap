// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartAddressInput should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StartAddressInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "className": "start-address-input",
      "debounce": 200,
      "highlightFirstSuggestion": false,
      "name": "startAddress",
      "onChange": [Function],
      "onError": [Function],
      "onSelect": [Function],
      "searchOptions": Object {},
      "shouldFetchSuggestions": true,
      "value": "",
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "className": "start-address-input",
        "debounce": 200,
        "highlightFirstSuggestion": false,
        "name": "startAddress",
        "onChange": [Function],
        "onError": [Function],
        "onSelect": [Function],
        "searchOptions": Object {},
        "shouldFetchSuggestions": true,
        "value": "",
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
